### PR TITLE
docs(caddy): update examples, simplify config

### PR DESCRIPTION
**Pull request by coolaj86**
Re: https://github.com/webinstall/webi-installers/issues/487

### CODE REVIEW

**Review by ryanburnette — APPROVED** [review]

**Comment by coolaj86** [comment]
Comment. Do you have any feedback @mholt?

Preview:
https://github.com/webinstall/webi-installers/tree/docs-caddy-handle/caddy

The goal is to provide the fewest examples possible, introducing the fewest directives to learn, in the most consistent way (better to always use `handle` even with `/` than to _sometimes_ use it), with real-world-ready examples.

File: caddy/README.md [hunk 1]
```diff
@@ -29,55 +29,112 @@ Here's the things we find most useful:
...
 ```
 
+### How to serve HTTPS on localhost
+
+Caddy can be used to test with https on localhost.
+
+`Caddyfile`:
+
+```Caddyfile
+https://localhost:443 {
```
Comment (mholt) [inline caddy/README.md:39]:
```suggestion
localhost {
```

File: caddy/README.md [hunk 2]
```diff
@@ -29,55 +29,112 @@ Here's the things we find most useful:
...
+    handle /api/* {
+        reverse_proxy localhost:3000
+    }
+    
+    handle /* {
```
Comment (mholt) [inline caddy/README.md:44]:
```suggestion
    handle {
```

File: caddy/README.md [hunk 3]
```diff
@@ -29,55 +29,112 @@ Here's the things we find most useful:
...
+        rewrite * /api/newpath{path}
+        reverse_proxy localhost:3000
+    }
+    
+    ###############
+    # File Server #
+    ###############
+    
+    # serve static files
+    handle /* {
```
Comment (mholt) [inline caddy/README.md:117]:
```suggestion
    handle {
```

File: caddy/README.md [hunk 4]
```diff
@@ -29,55 +29,112 @@ Here's the things we find most useful:
...
+    
+    handle /* {
+        root * ./public/
+        file_server
+    }
+}
+```
+
+```sh
+caddyfile run --config ./Caddyfile
```
Comment (mholt) [inline caddy/README.md:52]:
```suggestion
caddy run
```

**Review by mholt — COMMENTED** [review]
LGTM! These would be my recommendations for a simpler, less noisy config, but if you intentionally want to keep things the way they are for some reason, you are allowed to I guess.

Generally we discourage explicitly defining behavior that is already the default, since Caddy has sensible defaults, and it also makes the software unnecessarily complicated to learn and use especially when going from examples. So this is my take, i.e. I'd prefer this or do it this way if it was me writing it.

File: caddy/README.md (hunk 1 above)
Comment (coolaj86) [reply caddy/README.md:39]: I've had issues with this. I believe it had something to do with browser caching / redirects.

I've found only running localhost on 443 to be more reliable than running on 80 and 443.

File: caddy/README.md (hunk 2 above)
Comment (coolaj86) [reply caddy/README.md:44]: Too much mental load to have multiple ways to do the same thing.

There should be 1 obvious way to do it.

File: caddy/README.md (hunk 4 above)
Comment (coolaj86) [reply caddy/README.md:52]: Same as above. Too much mental load to have multiple ways to do the same thing, and to have the most common option hidden away.

**Comment by coolaj86** [comment]
@mholt I prefer the Go / Python / Zig mantras of "explicit over implicit", "Clear is better than clever", "1 obvious way to do it".

I want the lowest possible mental burden for people who are coming to webi, even if that means a few more characters to copy and paste.

That aside, are there other "every web developer is going to need to use this" type of options that should be included in the cheat sheet?

The [last one](https://webinstall.dev/caddy) was waaay to complicated, but I didn't know that there were simpler ways to do the same thing.

File: caddy/README.md (hunk 1 above)
Comment (mholt) [reply caddy/README.md:39]: These two are exactly the same though. Specifying `https://` still does redirects from HTTP, and serves on port 443 by default, just as the hostname alone does.

Comment (coolaj86) [reply caddy/README.md:39]: Noted. I'll switch it back.

**Pull request merged**